core/src/main/java/net/adoptopenjdk/icedteaweb/client/controlpanel/ControlPanel.java

```diff
@@ -241,130 +241,69 @@ public void actionPerformed(ActionEvent e) {
-                new SettingsPanel(Translator.R("APPEXTSECControlPanelExtendedAppletSecurityTitle"), new UnsignedAppletsTrustingListPanel(PathsAndFiles.APPLET_TRUST_SETTINGS_SYS.getFile(), PathsAndFiles.APPLET_TRUST_SETTINGS_USER.getFile(), this.config))
-        };
-
-        // Add panels.
-        final JPanel settingsPanel = new JPanel(new CardLayout());
-
-        // Calculate largest minimum size we should use.
-        int height = 0;
-        int width = 0;
-        for (SettingsPanel panel : panels) {
-            JPanel p = panel.getPanel();
-            Dimension d = p.getMinimumSize();
-            if (d.height > height) {
-                height = d.height;
-            }
-            if (d.width > width) {
-                width = d.width;
+        final List<ControlPanelProvider> providers = new ArrayList<>();
+        final ServiceLoader<ControlPanelProvider> serviceLoader = ServiceLoader.load(ControlPanelProvider.class);
+        serviceLoader.iterator().forEachRemaining(p -> {
+            final String name = p.getName();
+            if (p.isActive()) {
+                if (providers.stream().filter(provider -> Objects.equals(provider.getName(), name)).findAny().isPresent()) {
+                    throw new IllegalStateException("More than 1 view provider for control panel with name " + name + " found!");
+                }
+                LOG.debug("Adding view {} to control panel", name);
+                providers.add(p);
+            } else {
+                LOG.debug("Won't add view {} to control panel since it is deactivated", name);
             }
-        }
-        Dimension dim = new Dimension(width, height);
+        });
 
-        for (SettingsPanel panel : panels) {
-            JPanel p = panel.getPanel();
-            p.setPreferredSize(dim);
-            settingsPanel.add(p, panel.toString());
-        }
 
-        final JList<SettingsPanel> settingsList = new JList<>(panels);
-        settingsList.addListSelectionListener(new ListSelectionListener() {
-            @Override
-            public void valueChanged(ListSelectionEvent e) {
-                @SuppressWarnings("unchecked")
-                JList<SettingsPanel> list = (JList<SettingsPanel>) e.getSource();
-                SettingsPanel panel = list.getSelectedValue();
-                CardLayout cl = (CardLayout) settingsPanel.getLayout();
-                cl.show(settingsPanel, panel.toString());
-            }
-        });
-        JScrollPane settingsListScrollPane = new JScrollPane(settingsList);
-        settingsListScrollPane.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+        final Map<JComponent, String> panels = new HashMap<>();
+        final List<String> orderedNames = providers.stream().sorted(Comparator.comparingInt(p -> p.getOrder())).map(p -> p.getName()).collect(Collectors.toList());
+        providers.stream().forEach(p -> panels.put(p.createPanel(config), p.getName()));
```

Comment:
There is a `toMap` collector which allows to convert a stream into a map

Reply:
done